managers/restore-manager/RestoreTaskPoller.js

```diff
@@ -29,30 +30,33 @@ class RestoreTaskPoller {
       return RestoreService.createService(plan)
         .then(restoreService => restoreService
           .getRestoreOperationState(restore_opts)
-          .catch(e => logger.error('Caught error in poller', e))
           .then(operationStatusResponse => {
             logger.info(`Got restore operation response for guid ${changedOptions.restore_guid}`, operationStatusResponse);
-            if (_.includes([CONST.APISERVER.RESOURCE_STATE.SUCCEEDED, CONST.APISERVER.RESOURCE_STATE.FAILED], operationStatusResponse.state)) {
-              // cancel the poller and clear the array
-              clearInterval(interval);
-              RestoreTaskPoller.pollers[object.metadata.name] = false;
-            }
             return Promise
-              .try(() => {
-                if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
-                  return this._logEvent(restore_opts, CONST.OPERATION_TYPE.RESTORE, operationStatusResponse);
-                }
-              })
-              .then(() => eventmesh.apiServerClient.patchResource({
+              .try(() => eventmesh.apiServerClient.patchResource({
                 resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.RESTORE,
                 resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_RESTORE,
                 resourceId: changedOptions.restore_guid,
                 status: {
                   'state': operationStatusResponse.state,
                   response: operationStatusResponse
                 }
-              }));
+              }))
+              .then(() => {
+                if (_.includes([CONST.APISERVER.RESOURCE_STATE.SUCCEEDED, CONST.APISERVER.RESOURCE_STATE.ABORTED], operationStatusResponse.state)) {
+                  logger.debug('Cancel the poller:', object.metadata.name);
+                  clearInterval(interval);
+                  RestoreTaskPoller.pollers[object.metadata.name] = false;
+                }
+              })
+              .then(() => {
+                if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+                  _.set(restore_opts, 'user.name', changedOptions.username);
+                  return RestoreTaskPoller._logEvent(restore_opts, CONST.OPERATION_TYPE.RESTORE, operationStatusResponse);
+                }
+              });
           })
+          .catch(e => logger.error('Caught error in poller', e))
```

Comment:
Same here.

Reply:
On second thought, we can cancel the poller here. and whenever watch refreshes we will anyway get the event again hence new poller will start. We will avoid running erroneous poll every 3 sec rather it will be done every 1 minute when watch refreshes

Reply:
What happens if this error comes in [`_logEvent`](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/381/files#diff-8c64b223c0a29cc710c7f01ce1d89d8dR55)
State is succeeded but poller won't be cancelled, makes sense to cancel the poller => next event won't come.

Reply:
The poller is already cancelled at line 48